core/src/main/java/org/kiwix/kiwixmobile/core/reader/ZimFileReader.kt

```diff
@@ -98,8 +99,14 @@ class ZimFileReader(
   fun searchSuggestions(prefix: String, count: Int) =
     jniKiwixReader.searchSuggestions(prefix, count)
 
-  fun getNextSuggestion(): String? =
-    valueOfJniStringAfter(jniKiwixReader::getNextSuggestion)
+  fun getNextSuggestion(): SearchSuggestion? {
+    val title = JNIKiwixString()
+    val url = JNIKiwixString()
+    if (jniKiwixReader.getNextSuggestion(title, url)) {
```

Comment:
it has my approval but you could return the if here instead of having 2 returns
```
return if (jniKiwixReader.getNextSuggestion(title, url)) SearchSuggestion(title.value, url.value)
       else null
```
kotlin lacks a specific ternary operator so you have to use if/else. In these scenarios I omit the braces

Reply:
Made the change [here](https://github.com/kiwix/kiwix-android/pull/1537/commits/c2412dee6c2569136d1beb77a9435b6600625f82)
I was getting a style warning on indenting the `else null`, so left it at the beginning